src/Microsoft.AspNet.Server.Kestrel/ServerFactory.cs

```diff
@@ -63,18 +65,33 @@ public IDisposable Start(IFeatureCollection serverFeatures, Func<IFeatureCollect
                 }
 
                 engine.Start(information.ThreadCount == 0 ? 1 : information.ThreadCount);
+                bool atLeastOneListener = false;
 
                 foreach (var address in information.Addresses)
                 {
-                    disposables.Push(engine.CreateServer(
-                        address.Scheme,
-                        address.Host,
-                        address.Port,
-                        async frame =>
-                        {
-                            var request = new ServerRequest(frame);
-                            await application.Invoke(request.Features).ConfigureAwait(false);
-                        }));
+                    var parsedAddress = ServerAddress.FromUrl(address);
+                    if (parsedAddress == null)
+                    {
+                        _logger.LogError("Unrecognized listening address: " + address);
```

Comment:
Remove the address from information.Addresses?

That way this following won't be a lie: https://github.com/aspnet/Hosting/blob/40c82d54d254e633fa20c3231014ca7ba2e0c8c6/src/Microsoft.AspNet.Hosting/Program.cs#L50